test/test_point_inclusion.cpp

```diff
@@ -211,6 +211,50 @@ TEST(BoxRayIntersection, SimpleRay1)
     EXPECT_TRUE(intersect);
 }
 
+TEST(BoxRayIntersection, BrokenRay)
```

Comment:
Can you find more descriptive names for the unittests?

Reply:
I dont know what caused the failures, so any suggestions? NonSimpleRay1&2 ?
> the tests are just random failed cases. Best case would be to investigate why the rays did not work

Reply:
What about SimpleRay2&3 then?